File: frontend/src/types/job.ts
export interface Job {
  id: string
  project_id: string | null
  job_type: string
  status: 'queued' | 'processing' | 'done' | 'error' | 'cancelled'
  progress: number
  params_json: string | null
  result_json: string | null
  error_message: string | null
  created_at: string
  started_at: string | null
  completed_at: string | null
}

File: backend/app/schemas/job.py
from datetime import datetime

from pydantic import BaseModel


class JobResponse(BaseModel):
    id: str
    project_id: str | None = None
    job_type: str
    status: str
    progress: float
    params_json: str | None = None
    result_json: str | None = None
    error_message: str | None = None
    created_at: datetime
    started_at: datetime | None = None
    completed_at: datetime | None = None

    model_config = {"from_attributes": True}
